### PR TITLE
roachtest: allow for overriding mvt planner params through env vars

### PR DESCRIPTION
**Pull request by DarrylWong**
When debugging mixed version tests, one often wishes to generate a specific
upgrade path or use a specific deployment mode. The latter is not possible
without changing the source code while the former is not supported at all.

This change exposes two env vars, MVT_UPGRADE_PATH and MVT_DEPLOYMENT_MODE
to allow for overriding mixed version plan generation.

Release note: none
Epic: none
Fixes: none

### CODE REVIEW

File: pkg/cmd/roachtest/roachtestutil/mixedversion/mixedversion.go [hunk 1]
```diff
@@ -839,6 +864,7 @@ func (t *Test) runCommandFunc(nodes option.NodeListOption, cmd string) stepFunc
 // only available on v22.2.0+.
 func (t *Test) choosePreviousReleases() ([]*clusterupgrade.Version, error) {
 	skipVersions := t.prng.Float64() < t.options.skipVersionProbability
+
```
Comment (srosenberg) [inline pkg/cmd/roachtest/roachtestutil/mixedversion/mixedversion.go:867]: Nit: unintended or you really prefer a line break here? :)

File: pkg/cmd/roachtest/roachtestutil/mixedversion/mixedversion.go [hunk 2]
```diff
@@ -1324,3 +1359,38 @@ func assertValidTest(test *Test, fatalFunc func(...interface{})) {
...
+		// care about the exact patch version.
+		parsedVersion, err = clusterupgrade.LatestPatchRelease(v)
+		if err != nil {
+			return nil, errors.Newf("unable to parse version: %s", v)
+		}
+
+		upgradePath = append(upgradePath, parsedVersion)
+	}
+
+	return upgradePath, nil
```
Comment (srosenberg) [inline pkg/cmd/roachtest/roachtestutil/mixedversion/mixedversion.go:1395]: We should log success in overriding the upgrade path, as is done on L963.

File: pkg/cmd/roachtest/roachtestutil/clusterupgrade/clusterupgrade.go [hunk 3]
```diff
@@ -101,19 +101,49 @@ func CurrentVersion() *Version {
...
+		return nil, err
+	}
+
+	return &Version{*parsedVersion}, nil
+}
+
+// LatestPatchRelease returns the latest patch release version for a given
+// release series.
+func LatestPatchRelease(series string) (*Version, error) {
+	seriesStr := strings.TrimPrefix(series, "v")
```
Comment (srosenberg) [inline pkg/cmd/roachtest/roachtestutil/clusterupgrade/clusterupgrade.go:138]: The ping-pong around the "v" prefix is amusing.

**Review by srosenberg — APPROVED** [review]

File: pkg/cmd/roachtest/roachtestutil/mixedversion/mixedversion.go (hunk 1 above)
Comment (DarrylWong) [reply pkg/cmd/roachtest/roachtestutil/mixedversion/mixedversion.go]: Fixed :)

File: pkg/cmd/roachtest/roachtestutil/mixedversion/mixedversion.go (hunk 2 above)
Comment (DarrylWong) [reply pkg/cmd/roachtest/roachtestutil/mixedversion/mixedversion.go:1395]: Done.

**Comment by DarrylWong** [comment]
TFTR!

bors r=srosenberg

**Pull request merged**